### PR TITLE
Journal

### PR DESCRIPTION
**Pull request by symbiont-stevan-andjelkovic**

### CODE REVIEW

File: src/journal/test/JournalTest.hs [hunk 1]
```diff
@@ -78,18 +78,79 @@ appendBSFake bs fj@(FakeJournal bss ix termCount) =
...
+    termLen :: Int
+    termLen = oTermBufferLength testOptions
+
+    position = readBytes + unreadBytes
+
+    limit :: Int
+    limit = readBytes + termLen `div` 2
+
+    hardLimit :: Int
+    hardLimit = readBytes + termLen -- TODO: calculate this correctly
```
Comment (symbiont-daniel-gustafsson) [inline src/journal/test/JournalTest.hs:107]: can be removed

File: src/journal/test/JournalTest.hs [hunk 2]
```diff
@@ -78,18 +78,79 @@ appendBSFake bs fj@(FakeJournal bss ix termCount) =
...
-                           else Vector.snoc bss padding) ix (termCount + 1), Left Rotation)
-      else (FakeJournal (Vector.snoc bss bs) ix termCount, Right ())
+      then if journalLength < termLen * termCount
+           then [(FakeJournal (if BS.length padding == 0
+                             then bss
+                             else Vector.snoc bss padding) ix (termCount + 1), Left Rotation)]
+           else [(FakeJournal bss ix (termCount + 1), Left Rotation)]
+      else if journalLength' > termLen * (min 1 (termCount - 1))
+           then [ (FakeJournal (Vector.snoc bss bs) ix termCount, Right ())
+                , (FakeJournal bss ix (termCount + 1), Left Rotation)
```
Comment (symbiont-daniel-gustafsson) [inline src/journal/test/JournalTest.hs:151]: Should it really be a `termCount + 1` here (and same on line 148? I'm note sure I completely understand the different conditions in play here.

**Review by symbiont-daniel-gustafsson — APPROVED** [review]

File: src/journal/test/JournalTest.hs (hunk 1 above)
Comment (symbiont-stevan-andjelkovic) [reply src/journal/test/JournalTest.hs]: Done, thanks!

File: src/journal/test/JournalTest.hs (hunk 2 above)
Comment (symbiont-stevan-andjelkovic) [reply src/journal/test/JournalTest.hs]: I think you are right, fixed, thanks!

**Pull request merged**